x/lazyledgerapp/types/payformessage.go

```diff
@@ -0,0 +1,294 @@
+package types
+
+import (
+	"crypto/sha256"
+	"errors"
+	fmt "fmt"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lazyledger/lazyledger-core/crypto/merkle"
+	core "github.com/lazyledger/lazyledger-core/types"
+	"github.com/lazyledger/nmt"
+)
+
+const (
+	TypeMsgPayforMessage                   = "payformessage"
+	TypeSignedTransactionDataPayForMessage = "signedtransactiondatapayformessage"
+	ShareSize                              = core.ShareSize
+	SquareSize                             = 64
+	NamespaceIDSize                        = core.NamespaceSize
+)
+
+///////////////////////////////////////
+// 	MsgWirePayForMessage
+///////////////////////////////////////
+
+var _ sdk.MsgRequest = &MsgWirePayForMessage{}
+
+func (msg *MsgWirePayForMessage) Route() string { return RouterKey }
+
+func (msg *MsgWirePayForMessage) Type() string { return TypeMsgPayforMessage }
+
+// ValidateBasic checks for valid namespace length, declared message size, share
+// commitments, signatures for those share commitments, and fulfills the sdk.Msg
+// interface
+func (msg *MsgWirePayForMessage) ValidateBasic() error {
+	pubK := msg.PubKey()
+
+	// ensure that the namespace id is of length == 8
+	if len(msg.GetMessageNameSpaceId()) != NamespaceIDSize {
+		return fmt.Errorf(
+			"invalid namespace length: got %d wanted 8",
```

Comment:
Use constant here instead of hardcoded 8 just for future maintenance.

Reply:
this is resolved in https://github.com/lazyledger/lazyledger-app/pull/21/commits/991c8a58fc0c64c2ae668ade8dd383365be94151 and https://github.com/lazyledger/lazyledger-app/pull/21/commits/72ab5f921270b975ef5908e7e35faebc19dc4dac

Reply:
For posterity: https://github.com/lazyledger/lazyledger-app/pull/21#discussion_r579658156